samples/assistant/csharp-ooproc/README.md

```diff
@@ -0,0 +1,280 @@
+# Assistants
+
+This sample demonstrates how to build an AI assistant chatbot with custom skills using Azure Functions and a local build of the OpenAI extension.
+It builds upon the concepts [chatbot](../chatbot) sample, which demonstrates how to create a simple chatbot.
+
+## Introduction
+
+AI assistants are chat bots that can be configured with custom skills. Custom skills are implemented using the `assistantSkillTrigger` binding.
+Custom skills are useful ways to extend the functionality of an AI assistant. For example, you can create a custom skill that saves a todo item to a database, or a custom skill that queries a database for a list of todo items.
+In both cases, the skills are defined in your app and any the language model doesn't need to know anything about the skill implementation, making it useful for interacting with private data.
+
+This OpenAI extension internally uses the [function calling](https://platform.openai.com/docs/guides/function-calling) functionality available in
+`gpt-3.5-turbo` and `gpt-4` models to implement assistant skills.
+
+## Defining skills
+
+You can define a skill by creating a function that uses the `AssistantSkillTrigger` binding. The following example shows a skill that adds a todo item to a database:
+
+C# example:
+
+```csharp
+[Function(nameof(AddTodo))]
+public Task AddTodo([AssistantSkillTrigger("Create a new todo task")] string taskDescription)
+{
+    if (string.IsNullOrEmpty(taskDescription))
+    {
+        throw new ArgumentException("Task description cannot be empty");
+    }
+
+    this.logger.LogInformation("Adding todo: {task}", taskDescription);
+
+    string todoId = Guid.NewGuid().ToString()[..6];
+    return this.todoManager.AddTodoAsync(new TodoItem(todoId, taskDescription));
+}
+```
+
+Nodejs example:
+
+```ts
+app.generic('AddTodo', {
+    trigger: trigger.generic({
+        type: 'assistantSkillTrigger',
+        functionDescription: 'Create a new todo task'
+    }),
+    handler: async (taskDescription: string, context: InvocationContext) => {
+        if (!taskDescription) {
+            throw new Error('Task description cannot be empty')
+        }
+
+        context.log(`Adding todo: ${taskDescription}`)
+
+        const todoId = crypto.randomUUID().substring(0, 6)
+        return todoManager.AddTodo(new TodoItem(todoId, taskDescription))
+    }
+})
+```
+
+Python example:
```

Comment:
This README file is in a C# sample, so there's no benefit to including Python examples. Rather, python examples should be added to the higher-level `/samples/assistant/README.md` file.

Reply:
it was mistakenly copied in csharp folder, deleted now